crates/prune/src/pruner.rs

```diff
@@ -253,13 +263,150 @@ impl<DB: Database> Pruner<DB> {
+        // Contract log filtering removes every receipt possible except the ones in the list. So,
+        // for the other receipts it's as if they had a `PruneMode::Distance()` of 128.
+        let to_block = PruneMode::Distance(MINIMUM_PRUNING_DISTANCE)
+            .prune_target_block(
+                tip_block_number,
+                MINIMUM_PRUNING_DISTANCE,
+                PrunePart::ContractLogs,
+            )?
+            .map(|(bn, _)| bn)
+            .unwrap_or_default();
+
+        // Figure out what receipts have already been pruned, so we can have an accurate
+        // `address_filter`
+        let pruned = provider
+            .get_prune_checkpoint(PrunePart::ContractLogs)?
+            .map(|checkpoint| checkpoint.block_number)
+            .unwrap_or(0);
+
+        let address_filter =
+            self.modes.contract_logs_filter.group_by_block(tip_block_number, Some(pruned))?;
+
+        // Split all transactions in different block ranges. Each block range will have its own
+        // filter address list.
+        let mut block_ranges = vec![];
+        let mut blocks_iter = address_filter.keys().peekable();
+        let mut filtered_addresses = vec![];
+
+        while let Some(start_block) = blocks_iter.next() {
+            filtered_addresses.extend_from_slice(address_filter.get(start_block).expect("qed"));
+
+            // This will clear all receipts before the first  appearance of a contract log
+            if block_ranges.is_empty() {
+                block_ranges.push((0, *start_block, 0));
+            }
+
+            let end_block =
+                blocks_iter.peek().map(|next_block| *next_block - 1).unwrap_or(to_block);
+
+            // Addresses in lower block ranges, are still included in the inclusion list for future
+            // ranges.
+            block_ranges.push((*start_block, end_block, filtered_addresses.len()));
+        }
+
+        for (start_block, end_block, num_addresses) in block_ranges {
+            let range = match self.get_next_tx_num_range_from_checkpoint(
+                provider,
+                PrunePart::ContractLogs,
+                end_block,
+            )? {
+                Some(range) => range,
+                None => {
+                    trace!(target: "pruner", "No receipts to filter and prune between {start_block} and {end_block} blocks.");
+                    continue
+                }
+            };
+
+            let total = range.clone().count();
+            let mut processed = 0;
+
+            provider.prune_table_with_iterator_in_batches::<tables::Receipts>(
+                range,
+                self.batch_sizes.receipts,
+                |rows| {
+                    processed += rows;
+                    trace!(
+                        target: "pruner",
+                        %rows,
+                        progress = format!("{:.1}%", 100.0 * processed as f64 / total as f64),
+                        "Filtered or pruned receipts"
```

Comment:
1. I think we're not actually filtering anything inside receipts, so it's just pruned
2. We can specify the block range we're pruning, as the log will be written multiple times
```suggestion
                        "Pruned receipts for block range"
```